modmesh/apputil.py

```diff
@@ -57,46 +58,68 @@ class AppEnvironment:
     :ivar locals:
         The local namespace of the application.
     """
-    def __init__(self, name):
+
+    # default config
+    _config = {
+        'appEnvName': None,
+        'redirectStdOutFile': 'stdout.txt',
+        'redirectStdErrFile': 'stderr.txt',
+    }
+
+    def __init__(self, config=None):
         self.globals = {
             # Give the application an alias of the top package.
             'mm': importlib.import_module('modmesh'),
             'appenv': self,
         }
         self.locals = {}
-        self.name = name
         # Each run of the application appends a new environment.
-        environ[name] = self
+        environ[config['appEnvName']] = self
 
-    def run_code(self, code):
-        exec(code, self.globals, self.locals)
+        # replace the config value; otherwise keep the default value
+        for k, v in config.items():
+            if v is not None:
+                self._config[k] = v
 
-
-def get_appenv(name=None):
-    if None is name:
+    def run_code(self, code):
+        with open(self._config['redirectStdOutFile'], 'w') as f1:
+            with contextlib.redirect_stdout(f1):
+                with open(self._config['redirectStdErrFile'], 'w') as f2:
+                    with contextlib.redirect_stderr(f2):
+                        try:
+                            exec(code, self.globals, self.locals)
+                        except Exception as e:
+                            print(("{}: {}".format(
+                                type(e).__name__, str(e))), file=sys.stderr)
+                            print("traceback:", file=sys.stderr)
+                            traceback.print_stack()
+                        sys.stdout.flush()
+                        sys.stderr.flush()
+
+
+def new_appenv(config=None):
+    name = config.get('appEnvName', None)
+    if None is name or name == '':
```

Comment:
```suggestion
    if name is None or name == '':
```

Reply:
For making the two conditions in the line use consistent style, changing the order is good for reading.

If not in this scenario, (like a single condition), I prefer putting constant in the first place in the conditional checking.  It is the same as a convention commonplace in C code, which is to reduce the chance of typo to treat equality `==` as assignment `=`, because

```cpp
if (nullptr = pointer)
```

is an error compiler must complain about.

With the aid of modern compilers or clang-tidy, the trick is not absolutely necessary, but good to have.